### PR TITLE
LPS-169220 No error message is displayed When a validation is not valid

### PR DESCRIPTION
**Pull request by diogomelo09**
[LPS-169220](https://issues.liferay.com/browse/LPS-169220)

### CODE REVIEW

**Comment by diogomelo09** [comment]
ci:reevaluate:1267401_7259

**Comment by diogomelo09** [comment]
ci:reevaluate:1387401_5983

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java [hunk 1]
```diff
@@ -579,6 +582,15 @@ private void _initSystemObjectDefinition(
...
 	private GroupLocalService _groupLocalService;
 
+	@Reference
+	private JSONFactory _jsonFactory;
+
+	@Reference
+	private Language _language;
+
+	@Reference
+	private ObjectActionEngine _objectActionEngine;
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java:592]: @diogomelo09, are we using `_objectActionEngine` and `_jsonFactory` in this class?

File: modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleEngineException.java [hunk 2]
```diff
@@ -21,21 +21,66 @@
...
 
-	public ObjectValidationRuleEngineException(String msg) {
-		super(msg);
+	public String getMessageKey() {
+		return _messageKey;
 	}
 
-	public ObjectValidationRuleEngineException(
-		String msg, Throwable throwable) {
+	public static class InvalidEngine
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleEngineException.java:32]: @diogomelo09, please rename `InvalidEngine` to `NoSuchEngine`.

File: modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleEngineException.java [hunk 3]
```diff
@@ -21,21 +21,66 @@
...
+
+		public InvalidScript() {
+			super(
+				"There was an error validating your data.",
+				"there-was-an-error-validating-your-data");
+		}
+
 	}
 
+	public static class NullEngine extends ObjectValidationRuleEngineException {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectValidationRuleEngineException.java:61]: Rename `NullEngine` to `MustNotBeNull`

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/ObjectValidationRuleEngineExceptionMapper.java [hunk 4]
```diff
@@ -26,14 +29,39 @@
...
+				_language.get(
+					_acceptLanguage.getPreferredLocale(),
+					objectValidationRuleEngineException.getMessage()));
+			else
+
+			return new Problem(
+				Response.Status.BAD_REQUEST,
+				_language.get(
+					_acceptLanguage.getPreferredLocale(),
+					objectValidationRuleEngineException.getMessageKey()));
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/ObjectValidationRuleEngineExceptionMapper.java:54]: @diogomelo09, I believe this approach below is more simple, what do you think?

````
String messageKey = objectValidationRuleEngineException.getMessageKey();

if (messageKey == null) {
	messageKey = objectValidationRuleEngineException.getMessage();
}

return new Problem(
	Response.Status.BAD_REQUEST,
	_language.get(
		_acceptLanguage.getPreferredLocale(), messageKey));
````

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
@diogomelo09,

I ask for some changes, feel free to contact me or @p-albuquerque if there is doubts.

Best Regards,
Feliphe

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/ObjectValidationRuleEngineExceptionMapper.java (hunk 4 above)
Comment (diogomelo09) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/exception/mapper/ObjectValidationRuleEngineExceptionMapper.java]: Just did it!! :smile:

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by diogomelo09** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
@diogomelo09,

could you please take a look at these failures:

[CreateObject#CannotUpdateObjectNameAfterPublished](https://testray.liferay.com/home/-/testray/case_results/2178073131?tab=history)
[ObjectFields#CanViewDocumentsAndMediaOption](https://testray.liferay.com/home/-/testray/case_results/2178073493/history?tab=result&hideFilterPin=true)
[ObjectRelationships#CannotRelateAnEntryWithItself](https://testray.liferay.com/home/-/testray/case_results/2178073628/history?tab=result&hideFilterPin=true)
[CreateObject#CanAddRelationOnRelationshipTab](https://testray.liferay.com/home/-/testray/case_results/2178073916/history?tab=result&hideFilterPin=true)

**Comment by evanilson-santana** [comment]
ci:test:object

**Comment by diogomelo09** [comment]
Resent here: #129561

**Pull request closed without merging**